### ISSUE TITLE
🐛 BUG: Importing Astro components from a NPM package ommits the styling inside the <style> tag

### ISSUE THREAD

**Issue by markteekman** [snowpackjs/astro#1204]
### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

I followed the documentations instructions on how to [Publish a Component](https://docs.astro.build/guides/publish-to-npm). The result is this  [Accessible Astro Components](https://www.npmjs.com/package/accessible-astro-components) package. Whenever I install this package on an Astro website everything works (props, JavaScript), all but the styling. It looks like nothing from within the `<style lang="scss">` tag from [AccordionItem.astro](https://github.com/markteekman/accessible-astro-components/blob/main/AccordionItem.astro) is loaded onto the page. This poses a problem as this components needs its initial styles to hide the content on it's panel, only to be triggered when a user clicks on an item.

I've had some suggestions from @jasikpark and @Tc-001 such as these two:
- Lazy loading components: https://github.com/snowpackjs/astro/pull/675#discussion_r669952553
- Check if theres' a `<head>` tag in the file: https://github.com/snowpackjs/astro/issues/1154

And by adding global to the tag: `<style global lang="scss">`, but none of these have worked so far.

This is the expected result:

![screenshot-expected-result](https://user-images.githubusercontent.com/3909046/130504550-c0ef5f94-bb6e-4c04-a202-9daaa9d2dd10.JPG)

And this is what actually happens when adding the component via the package:

![screenshot-unexpected-result](https://user-images.githubusercontent.com/3909046/130504597-c16ed885-53b1-40b7-a2ce-e4121e3a3d54.JPG)

### Steps to Reproduce

1. `npm init astro` or run any Astro project
2. `npm install accessible-astro-components --save-dev`
3. Add the following to any Astro page:
```
---
import Accordion from 'accessible-astro-components/Accordion.astro'
import AccordionItem from 'accessible-astro-components/AccordionItem.astro'
---
<Accordion>
  <AccordionItem
    header="First Item"
  >
    <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto quasi nobis optio? Qui in quo accusantium debitis sapiente obcaecati magnam incidunt sit. Molestiae exercitationem quibusdam quod veritatis laboriosam est tenetur. </p>
    <a href="#">Tab to me!</a>
  </AccordionItem>
  <AccordionItem
    header="Second Item"
  >
    <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Architecto quasi nobis optio? Qui in quo accusantium debitis sapiente obcaecati magnam incidunt sit. Molestiae exercitationem quibusdam quod veritatis laboriosam est tenetur. </p>
  </AccordionItem>
  <!-- ... -->
</Accordion>
```
4. Error! No styling is applied (styles from the `<style>` tag in `AccordionItem.astro` should be loaded)

### Link to Minimal Reproducible Example (Optional)

https://accessible-astro.markteekman.nl/accordion/

**Comment by natemoo-re** [comment]
Hey @markteekman thanks for your patience! `astro@next` is in preview and has been merged to `main` here.

I just confirmed that `accessible-astro-components` are handled properly in the upcoming release! See [working reproduction](https://stackblitz.com/edit/github-y4db4y?file=src%2Fpages%2Findex.astro).

Going to leave this open until this is all released as stable.

**Comment by markteekman** [comment]
Awesome @natemoo-re, looks great! Thanks for the update on this one, looking forward to the next release! 😃

**Comment by natemoo-re** [comment]
Following up on my previous comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
